handler_test.go

```diff
@@ -1233,65 +1278,67 @@ func (s *handlerSuite) TestHandleErrorsWithErrorAfterWriteHeader(c *gc.C) {
+		c.Assert(rec.Body.String(), qt.Equals, "")
 	}
 }
 
-func (s *handlerSuite) TestHandleJSON(c *gc.C) {
+func TestHandleJSON(t *testing.T) {
+	c := qt.New(t)
+
 	req := new(http.Request)
 	params := httprouter.Params{}
 	// Test when handler returns an error.
 	handler := testServer.HandleJSON(func(p httprequest.Params) (interface{}, error) {
 		assertRequestEquals(c, p.Request, req)
-		c.Assert(p.PathVar, jc.DeepEquals, params)
-		c.Assert(p.PathPattern, gc.Equals, "")
+		c.Assert(p.PathVar, qt.DeepEquals, params)
+		c.Assert(p.PathPattern, qt.Equals, "")
 		return nil, errUnauth
 	})
 	rec := httptest.NewRecorder()
 	handler(rec, new(http.Request), params)
 	resp := parseErrorResponse(c, rec.Body.Bytes())
-	c.Assert(resp, gc.DeepEquals, &httprequest.RemoteError{
+	c.Assert(resp, qt.DeepEquals, &httprequest.RemoteError{
 		Message: errUnauth.Error(),
 		Code:    "unauthorized",
 	})
-	c.Assert(rec.Code, gc.Equals, http.StatusUnauthorized)
+	c.Assert(rec.Code, qt.Equals, http.StatusUnauthorized)
 
 	// Test when handler returns a body.
 	handler = testServer.HandleJSON(func(p httprequest.Params) (interface{}, error) {
 		assertRequestEquals(c, p.Request, req)
-		c.Assert(p.PathVar, jc.DeepEquals, params)
-		c.Assert(p.PathPattern, gc.Equals, "")
+		c.Assert(p.PathVar, qt.DeepEquals, params)
+		c.Assert(p.PathPattern, qt.Equals, "")
 		p.Response.Header().Set("Some-Header", "value")
 		return "something", nil
 	})
 	rec = httptest.NewRecorder()
 	handler(rec, req, params)
-	c.Assert(rec.Code, gc.Equals, http.StatusOK)
-	c.Assert(rec.Body.String(), gc.Equals, `"something"`)
-	c.Assert(rec.Header().Get("Some-Header"), gc.Equals, "value")
+	c.Assert(rec.Code, qt.Equals, http.StatusOK)
+	c.Assert(rec.Body.String(), qt.Equals, `"something"`)
+	c.Assert(rec.Header().Get("Some-Header"), qt.Equals, "value")
 }
 
-func assertRequestEquals(c *gc.C, req1, req2 *http.Request) {
-	c.Assert(req1.Method, gc.Equals, req2.Method)
-	c.Assert(req1.URL, jc.DeepEquals, req2.URL)
-	c.Assert(req1.Proto, gc.Equals, req2.Proto)
-	c.Assert(req1.ProtoMajor, gc.Equals, req2.ProtoMajor)
-	c.Assert(req1.ProtoMinor, gc.Equals, req2.ProtoMinor)
-	c.Assert(req1.Header, jc.DeepEquals, req2.Header)
-	c.Assert(req1.Body, gc.Equals, req2.Body)
-	c.Assert(req1.ContentLength, gc.Equals, req2.ContentLength)
-	c.Assert(req1.TransferEncoding, jc.DeepEquals, req2.TransferEncoding)
-	c.Assert(req1.Close, gc.Equals, req2.Close)
-	c.Assert(req1.Host, gc.Equals, req2.Host)
-	c.Assert(req1.Form, jc.DeepEquals, req2.Form)
-	c.Assert(req1.PostForm, jc.DeepEquals, req2.PostForm)
-	c.Assert(req1.MultipartForm, jc.DeepEquals, req2.MultipartForm)
-	c.Assert(req1.Trailer, jc.DeepEquals, req2.Trailer)
-	c.Assert(req1.RemoteAddr, gc.Equals, req2.RemoteAddr)
-	c.Assert(req1.RequestURI, gc.Equals, req2.RequestURI)
-	c.Assert(req1.TLS, gc.Equals, req2.TLS)
-	c.Assert(req1.Cancel, gc.Equals, req2.Cancel)
+func assertRequestEquals(c *qt.C, req1, req2 *http.Request) {
```

Comment:
This could probably be done less awkwardly now with cmpopts. Maybe something for later - perhaps leave a TODO?

Reply:
done